app/v1/models/request/ListBsasRequest.scala

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.models.request
+
+import uk.gov.hmrc.auth.core.Nino
+import v1.models.des.IncomeSourceIdentifier
+
+case class ListBsasRequest(nino: Nino, taxYear: DesTaxYear, incomeSourceIdentifier: Option[IncomeSourceIdentifier], identifierValue: Option[String])
```

Comment:
Either IncomeSourceIdentifier needs a toString method attached to each of the case objects, or this field should be made an Option[String] so that we can send it as a query parameter.